spring-batch-docs/asciidoc/scalability.adoc

```diff
@@ -157,76 +161,75 @@ See the section on <<step.adoc#split-flows,Split Flows>> for more
-    multiple remote processes. Clearly this pattern works best if the Master
+The master component is a single process, and the slaves are
+    multiple remote processes. This pattern works best if the master
     is not a bottleneck, so the processing must be more expensive than the
-    reading of items (this is often the case in practice).
+    reading of items (as is often the case in practice).
 
-The Master is just an implementation of a Spring Batch
-    Step, with the `ItemWriter` replaced with a generic
+The master is an implementation of a Spring Batch
+    `Step` with the `ItemWriter` replaced by a generic
     version that knows how to send chunks of items to the middleware as
-    messages. The Slaves are standard listeners for whatever middleware is
-    being used (e.g. with JMS they would be
-    `MesssageListeners`), and their role is to process
+    messages. The slaves are standard listeners for whatever middleware is
+    being used (for example, with JMS, they would be
+    `MesssageListener` implementations), and their role is to process
     the chunks of items using a standard `ItemWriter` or
     `ItemProcessor` plus
     `ItemWriter`, through the
     `ChunkProcessor` interface. One of the advantages of
-    using this pattern is that the reader, processor and writer components are
+    using this pattern is that the reader, processor, and writer components are
     off-the-shelf (the same as would be used for a local execution of the
     step). The items are divided up dynamically and work is shared through the
-    middleware, so if the listeners are all eager consumers, then load
+    middleware, so that, if the listeners are all eager consumers, then load
     balancing is automatic.
 
-The middleware has to be durable, with guaranteed delivery and
+The middleware has to be durable, with guaranteed delivery and a
     single consumer for each message. JMS is the obvious candidate, but other
-    options exist in the grid computing and shared memory product space (e.g.
-    Java Spaces).
+    options (such as Java Spaces exist in the grid computing and shared memory product space.
 
 [[partitioning]]
 
 
 === Partitioning
 
 Spring Batch also provides an SPI for partitioning a `Step` execution
-    and executing it remotely. In this case the remote participants are simply
+    and executing it remotely. In this case, the remote participants are
     `Step` instances that could just as easily have been configured and used for
-    local processing. Here is a picture of the pattern in action:
+    local processing. The following image shows the pattern:
 
 .Partitioning
 image::{batch-asciidoc}images/partitioning-overview.png[Partitioning Overview, scaledwidth="60%"]
 
-The `Job` is executing on the left hand side as a sequence of `Steps`,
-    and one of the `Steps` is labelled as a Master. The Slaves in this picture
+The `Job` runs on the left-hand side as a sequence of `Step` implementations,
```

Comment:
Wondering if instance(s) should be used instead of implementation(s)

I can change all of these on merge if you agree.

Reply:
Yes, please. I was thinking of the case where someone might make separate implementations to support different kinds of steps, but your idea makes more sense.